algo/src/test/java/com/hartwig/serve/sources/ckb/curation/CkbCuratorTest.java

```diff
@@ -1,33 +1,108 @@
+import com.hartwig.serve.ckb.datamodel.clinicaltrial.Location;
 import com.hartwig.serve.ckb.datamodel.variant.Variant;
 import com.hartwig.serve.sources.ckb.CkbTestFactory;
 
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 import org.junit.Test;
 
 public class CkbCuratorTest {
 
+    private final CkbCurator curator = CkbCuratorTestFactory.createProperModel();
+
     @Test
     public void canCurateVariants() {
-        CkbCurator curator = new CkbCurator();
-
-        CurationEntry firstCurationKey = CurationFactory.VARIANT_MAPPINGS.keySet().iterator().next();
+        CkbVariantCurationEntry firstCurationKey = CkbVariantCurationFactory.VARIANT_MAPPINGS.keySet().iterator().next();
         CkbEntry entry = CkbTestFactory.createEntryWithGeneAndVariant(firstCurationKey.geneSymbol(), firstCurationKey.variant());
 
         List<CkbEntry> entries = curator.run(Lists.newArrayList(entry));
 
         Variant firstVariant = entries.get(0).variants().get(0);
 
-        CurationEntry firstCuratedValue = CurationFactory.VARIANT_MAPPINGS.get(firstCurationKey);
+        CkbVariantCurationEntry firstCuratedValue = CkbVariantCurationFactory.VARIANT_MAPPINGS.get(firstCurationKey);
         assertEquals(firstCuratedValue.geneSymbol(), firstVariant.gene().geneSymbol());
         assertEquals(firstCuratedValue.variant(), firstVariant.variant());
 
         curator.reportUnusedCurationEntries();
     }
+
+    @Test
+    public void canCurateFacilityNameBasedOnName() {
+        Location location1 = createLocation("Rotterdam", "Erasmus MC", "3062 PA");
+        Location location2 = createLocation("Rotterdam", "Erasmus ziekenhuis in Rotterdam", "3062 PA");
+        Location curatedLocation = createLocation("Rotterdam", "EMC", "3062 PA");
+
+        assertEquals(curator.curateFacilityName(location1), curatedLocation);
+        assertEquals(curator.curateFacilityName(location2), curatedLocation);
+    }
+
+    @Test
+    public void canCurateFacilityNameBasedOnZip() {
+        Location locationWithZip1 = createLocation("Groningen", "Research site", "9713 GZ");
+        Location locationWithZip2 = createLocation("Groningen", "Research site", "9713");
+        Location locationWithoutZip = createLocation("Almere", "Flevo location 2", null);
+        Location curatedLocationWithZip1 = createLocation("Groningen", "UMCG", "9713 GZ");
+        Location curatedLocationWithZip2 = createLocation("Groningen", "UMCG", "9713");
+        Location curatedLocationWithoutZip = createLocation("Almere", "Flevoziekenhuis", null);
+
+        assertEquals(curator.curateFacilityName(locationWithZip1), curatedLocationWithZip1);
+        assertEquals(curator.curateFacilityName(locationWithZip2), curatedLocationWithZip2);
+        assertEquals(curator.curateFacilityName(locationWithoutZip), curatedLocationWithoutZip);
+    }
+
+    @Test
+    public void canCurateFacilityNameBasedOnFilter() {
```

Comment:
Is this test named properly? It seems like no curation happens here? Shouldn't it be something like "skip..."?